### PR TITLE
feat(ratelimit): Improve rate limiting

### PR DESCRIPTION
**Pull request by Khaledgarbaya**
the rate limit of the client now happens inside the create-http-client

### CODE REVIEW

File: lib/rate-limit.js [hunk 1]
```diff
@@ -7,17 +9,20 @@ export default function rateLimit (instance, maxRetry = 10) {
     if (!config) {
       return Promise.reject(error)
     }
-    config.attempt = config.attempt || 0
+    // in axios reponse will be availsble only if there a response from server
```
Comment (axe312ger) [inline lib/rate-limit.js]: `availsble` small typo

File: lib/rate-limit.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+let attempt = 0
```
Comment (axe312ger) [inline lib/rate-limit.js]: Does ist make sense to have the attempt counter for all requests together? Shouldn't it be per request?

File: lib/rate-limit.js [hunk 3]
```diff
@@ -29,11 +34,12 @@ export default function rateLimit (instance, maxRetry = 10) {
...
       }
 
-      var wait = rateLimitHeaderValue || Math.pow(2, config.attempt)
+      var wait = rateLimitHeaderValue || Math.pow(2, attempt)
       // convert to ms and add jitter
       wait *= Math.floor(1000 + ((Math.random() * 200) + 500))
-
-      console.log(`Error occured. Waiting for ${wait}ms before retrying....`)
+      console.log(`---------------------------------------------------------`)
+      console.log(`${error.response.status} Error occured. Waiting for ${wait} ms before retrying....`)
```
Comment (axe312ger) [inline lib/rate-limit.js]: Emoji? :neckbeard:

File: test/unit/create-http-client-test.js [hunk 4]
```diff
@@ -1,19 +1,31 @@
...
   createHttpClient(axios, {
     accessToken: 'clientAccessToken',
     space: 'clientSpaceId',
     defaultHostname: 'defaulthost'
   })
 
   t.equals(axios.create.args[0][0].baseURL, 'https://defaulthost:443/spaces/clientSpaceId/')
+  teardown()
   t.end()
 })
```
Comment (axe312ger) [inline test/unit/create-http-client-test.js]: What about a tests that sends some successful reqeusts with failing in between? To make sure all go out and are not blocked by the failing ones?

Comment (Khaledgarbaya) [inline test/unit/create-http-client-test.js]: That's already covered in the rate-limit-test

File: lib/rate-limit.js (hunk 2 above)
Comment (Khaledgarbaya) [inline lib/rate-limit.js]: we don't want to retry forever that's why we need to have attempt there and not per request

**Pull request merged**